### PR TITLE
Implemented routing

### PR DESCRIPTION
**Pull request by akshaynr**
Implemented Routing

Screenshot 2024-04-22 at 10 14 12 PM
Screenshot 2024-04-22 at 10 14 31 PM
Screenshot 2024-04-22 at 10 29 11 PM

### CODE REVIEW

File: movies-app/src/pages/MovieList.js [hunk 1]
```diff
@@ -1,62 +1,58 @@
...
+    const selectedGenreQueryParam = searchParams.get('genre');
+    const searchQueryParam = searchParams.get('query');
 
-    const onSearchSubmit = useCallback((query) => {
-        console.log(`Search Query: ${query}`);
-        setTitleSearchQuery(query)
-    }, []);
+    // useEffect(() => {
+    //     if (selectedGenreQueryParam) {
+    //         setSelectedGenre(selectedGenreQueryParam);
```
Comment (shubhamagarwal) [inline movies-app/src/pages/MovieList.js]: remove unused code

File: movies-app/src/components/MovieTile/MovieTile.js [hunk 2]
```diff
@@ -1,18 +1,27 @@
...
+    const navigate = useNavigate();
+    const [isLoading, responseData] = useFetch(getMovieDetailsByIdUrl(movieId));
+    const { id, title, release_date, poster_path, genres, vote_average, runtime, overview } = responseData || {};
     const releaseYear = new Date(release_date).getFullYear();
     const displayGenres = () => {
-        if(genres.length === 1) return genres[0];
-        else if(genres.length > 1){
-            const lastGenre = genres.pop();
-            return `${genres.join(', ')} & ${lastGenre}`;
+        if (genres) {
```
Comment (shubhamagarwal) [inline movies-app/src/components/MovieTile/MovieTile.js:13]: line 13 & 14 can be clubbed ?

File: movies-app/src/hooks/useFetch.js [hunk 3]
```diff
@@ -23,6 +23,7 @@ export const useFetch = (url, dependencies = []) => {
         };
 
         fetchData();
+        console.log('Dependences:', dependencies);
```
Comment (shubhamagarwal) [inline movies-app/src/hooks/useFetch.js]: remove console

Comment (akshaynr) [reply movies-app/src/hooks/useFetch.js]: > remove console

Removed console

File: movies-app/src/pages/MovieList.js (hunk 1 above)
Comment (akshaynr) [reply movies-app/src/pages/MovieList.js]: Removed unused code

File: movies-app/src/components/MovieTile/MovieTile.js (hunk 2 above)
Comment (akshaynr) [reply movies-app/src/components/MovieTile/MovieTile.js:13]: Updated loggic

**Review by shubhamagarwal — APPROVED** [review]

**Pull request merged**